src/databricks/sqlalchemy/dialect/__init__.py

```diff
@@ -0,0 +1,300 @@
+
+        Given a :class:`_engine.Connection`, a string
+        `table_name`, and an optional string `schema`, return column
+        information as a list of dictionaries with these keys:
+
+        name
+          the column's name
+
+        type
+          [sqlalchemy.types#TypeEngine]
+
+        nullable
+          boolean
+
+        default
+          the column's default value
+
+        autoincrement
+          boolean
+
+        sequence
+          a dictionary of the form
+              {'name' : str, 'start' :int, 'increment': int, 'minvalue': int,
+               'maxvalue': int, 'nominvalue': bool, 'nomaxvalue': bool,
+               'cycle': bool, 'cache': int, 'order': bool}
+
+        Additional column attributes may be present.
+        """
+
+        _type_map = {
+            "boolean": types.Boolean,
+            "smallint": types.SmallInteger,
+            "int": types.Integer,
+            "bigint": types.BigInteger,
+            "float": types.Float,
+            "double": types.Float,
+            "string": types.String,
+            "varchar": types.String,
+            "char": types.String,
+            "binary": types.String,
+            "array": types.String,
+            "map": types.String,
+            "struct": types.String,
+            "uniontype": types.String,
+            "decimal": DatabricksDecimal,
+            "timestamp": DatabricksTimestamp,
+            "date": DatabricksDate,
+        }
+
+        with self.get_driver_connection(
+            connection
+        )._dbapi_connection.dbapi_connection.cursor() as cur:
+            resp = cur.columns(
+                catalog_name=self.catalog,
+                schema_name=schema or self.schema,
+                table_name=table_name,
+            ).fetchall()
+
+        columns = []
+
+        for col in resp:
+
+            # Taken from PyHive. This removes added type info from decimals and maps
+            _col_type = re.search(r"^\w+", col.TYPE_NAME).group(0)
+            this_column = {
+                "name": col.COLUMN_NAME,
+                "type": _type_map[_col_type.lower()],
+                "nullable": bool(col.NULLABLE),
+                "default": col.COLUMN_DEF,
+                "autoincrement": False if col.IS_AUTO_INCREMENT == "NO" else True,
+            }
+            columns.append(this_column)
+
+        return columns
+
+    def get_pk_constraint(self, connection, table_name, schema=None, **kw):
+        """Return information about the primary key constraint on
+        table_name`.
+
+        Given a :class:`_engine.Connection`, a string
+        `table_name`, and an optional string `schema`, return primary
+        key information as a dictionary with these keys:
+
+        constrained_columns
+          a list of column names that make up the primary key
+
+        name
+          optional name of the primary key constraint.
+
+        """
+        # TODO: implement this behaviour
```

Comment:
is it better to return like there is no primary key or to return "not implemented" error? I may depend on how much this is used in real life and the impact it has.

Reply:
I will research this. e2e tests fail without this. I'll need to check if there is a way to tell SQLA at setup time whether to even check for primary keys (this is actually a request from one of our SA's internally).

Reply:
Any updates on this?

Reply:
As far as I can tell there's no way to disable SQLA enforcing Primary / Foreign key relations. For a preliminary release, the above implementation is workable (already in production for numerous partners / customers) and we'll have a follow-up ticket to properly pull these from `information_schema` in UC.